CMakeLists.txt

```diff
@@ -30,14 +32,26 @@ else ()
     set(INTEL TRUE)
   endif ()
 endif ()
+
 if (CMAKE_CXX_COMPILER_ID STREQUAL "Clang")
   # Ideally we should use -mbranches-within-32B-boundaries but it's not widely
   # available so at least align loops to prevent unrelated code changes from
   # affecting benchmark results.
   set(ALIGN_OPTIONS -mllvm -align-all-blocks=5)
-else ()
+elseif (CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
+  set(ALIGN_FUNCTION_BYTES 32)
+  set(ALIGN_OPTIONS -falign-functions=${ALIGN_FUNCTION_BYTES})
+elseif (CMAKE_CXX_COMPILER_ID STREQUAL "Intel")
   set(ALIGN_OPTIONS -Wa,-mbranches-within-32B-boundaries)
```

Comment:
This should be used for GCC too, `-falign-functions` is not enough.

Reply:
Oh, what version of GCC supports it ?
I tried with gcc-9 and it seems it was not working ?

Reply:
This is a pretty recent addition to mitigate Intel JCC bug so maybe gcc 10.

Reply:
So in order of preference:

- `-mbranches-within-32B-boundaries` 
- `-align-all-blocks=5`
- `-falign-functions=32` 

Or should I try to have `-mbranches-within-32B-boundaries` + `-falign-functions=32` when possible on gcc ?

Reply:
`-mbranches-within-32B-boundaries` should be preferred, the other two are just poor approximations.

> Or should I try to have -mbranches-within-32B-boundaries + -falign-functions=32 when possible on gcc ?

No, `-falign-functions=32` is not needed with `-mbranches-within-32B-boundaries`.